### DISCUSSION TITLE
commit problem

### DISCUSSION THREAD

**Discussion by Ruanan-Terra** [community/community#123198] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I tried to commit earlier on a project that contained a very large file (I didn't know about the file size limit), and after realizing the problem, I can no longer commit to the project, even after adding the file in question to the .gitignore or completely deleting it from my project, the error warning about the file size persists. I've already tried deleting my repository and creating a new one, and it didn't work. Can someone help me?

![Captura de tela 2024-05-09 204830](https://github.com/community/community/assets/134150706/63e2689f-235b-4113-9837-f10a97675bac)

**Comment by msi-end** [comment]
1. **Purge the cache:** Use the following commands to clear the Git cache:
`git rm --cached <file>`
Replace <file> with the name of the large file. This removes the file from the Git index (the staging area), but keeps it in your working directory.

2. **Revert changes:** If you've made any changes to your project after adding the large file to .gitignore or deleting it, you should revert those changes to ensure the large file is no longer included. You can use:
`git checkout .
`This command will revert all changes in your working directory to the last committed state.
3. **Re-Clone repository**: If all else fails, try cloning the repository into a new directory. Make sure the large file is not present in the repository before attempting any commits.

**Reply by Ruanan-Terra** [reply]
### I did the first two steps and it didn't work, so I created a new repository on github, copied my entire project to another folder excluding the heavy file, and tried to re-commit from scratch, in the new repository, but the exact same error continued .

**Follow the code from my previous comment**

`PS C:\Projeto Airbnb - ML> git init
Reinitialized existing Git repository in C:/Projeto Airbnb - ML/.git/
PS C:\Projeto Airbnb - ML> git remote rm origin
PS C:\Projeto Airbnb - ML> git init
Reinitialized existing Git repository in C:/Projeto Airbnb - ML/.git/
PS C:\Projeto Airbnb - ML> git rm -r --cached .
rm '.gitattributes'
rm '.gitignore'
rm 'Base_airbnb.parquet'
rm 'Deploy.py'
rm 'READMAP.md'
rm 'Solu├º├úo Airbnb Rio.ipynb'
rm 'dados_final.parquet'
PS C:\Projeto Airbnb - ML> git init
Reinitialized existing Git repository in C:/Projeto Airbnb - ML/.git/
PS C:\Projeto Airbnb - ML> git commit -m "first commit"
[main 20e7fcf] first commit
 7 files changed, 1796 deletions(-)
 delete mode 100644 .gitattributes
 delete mode 100644 .gitignore
 delete mode 100644 Base_airbnb.parquet
 delete mode 100644 Deploy.py
 delete mode 100644 READMAP.md
 delete mode 100644 "Solu\303\247\303\243o Airbnb Rio.ipynb"
 delete mode 100644 dados_final.parquet
PS C:\Projeto Airbnb - ML> git add .
warning: in the working copy of 'Solu├º├úo Airbnb Rio.ipynb', LF will be replaced by CRLF the next time Git touches it
PS C:\Projeto Airbnb - ML> git commit -m "first commit"
[main c93e93c] first commit
 7 files changed, 1796 insertions(+)
 create mode 100644 .gitattributes
 create mode 100644 .gitignore
 create mode 100644 Base_airbnb.parquet
 create mode 100644 Deploy.py
 create mode 100644 READMAP.md
 create mode 100644 "Solu\303\247\303\243o Airbnb Rio.ipynb"
 create mode 100644 dados_final.parquet
PS C:\Projeto Airbnb - ML> git branch -M main
PS C:\Projeto Airbnb - ML> git remote add origin https://github.com/Ruanan-Terra/Project-Airbnb---ML.git
PS C:\Projeto Airbnb - ML> git push -u origin main
Enumerating objects: 33, done.
Counting objects: 100% (33/33), done.
Delta compression using up to 8 threads
Compressing objects: 100% (30/30), done.
Writing objects: 100% (33/33), 237.07 MiB | 7.18 MiB/s, done.
Total 33 (delta 12), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (12/12), done.
remote: error: Trace: b5163c78e9912280094f36f5cecdc8ecf99ad6da7f8dcad0c01e22302f137040
remote: error: See https://gh.io/lfs for more information.
remote: error: File modelo.joblib is 1100.28 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/Ruanan-Terra/Project-Airbnb---ML.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/Ruanan-Terra/Project-Airbnb---ML.git'
PS C:\Projeto Airbnb - ML> git rm --cached modelo.joblib
fatal: pathspec 'modelo.joblib' did not match any files

_PS C:\Projeto Airbnb - ML> dir


    Diretório: C:\Projeto Airbnb - ML


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        09/05/2024     20:29             51 .gitattributes
-a----        10/05/2024     10:49           3316 .gitignore
-a----        21/04/2024     15:55       49181094 Base_airbnb.parquet
-a----        28/04/2024     16:24        9164355 dados_final.parquet
-a----        09/05/2024     17:00           1944 Deploy.py
-a----        09/05/2024     14:42            971 READMAP.md
-a----        09/05/2024     16:14         184029 Solução Airbnb Rio.ipynb_
`

**Comment by ebndev** [comment]
Hey @Ruanan-Terra, 

When you copied the project over to a new folder, did you remove the remote origin? It might still be connecting to the same repo that had the commit with the large file.

You can try reverting that commit as well.
